HidSharp/Platform/MacOS/MacHidDevice.cs

```diff
@@ -174,7 +177,69 @@ public override byte[] GetRawReportDescriptor()
 
         public override string GetDeviceString(int index)
         {
-            throw new NotSupportedException();
+            unsafe
+            {
+                IOUSBDevRequest request;
+                uint hidEntry = (uint) IORegistryEntryFromPath(0, ref _path);
+                uint interfaceEntry = 0;
+                uint deviceEntry = 0;
+                IntPtr iodev = IntPtr.Zero;
+                IntPtr usbdev = IntPtr.Zero;
+                var buffer = new byte[256];
+                if (hidEntry != 0)
+                {
+                    IORegistryEntryGetParentEntry(hidEntry, kIOServicePlane, ref interfaceEntry);
+                    if (interfaceEntry != 0)
+                    {
+                        IORegistryEntryGetParentEntry(interfaceEntry, kIOServicePlane, ref deviceEntry);
+                    }
+                }
+                if (deviceEntry != 0)
+                {
+                    Int32 score = 0;
+
+                    IOCreatePlugInInterfaceForService(deviceEntry,
+                        kIOUSBDeviceUserClientTypeID, kIOCFPlugInInterfaceID,
+                        ref iodev, ref score);
+                    if (iodev != null)
+                    {
+                        IntPtr devPtr = IntPtr.Zero;
+                        var deviceInterface = kIOUSBDeviceInterfaceGUID;
+
+                        Marshal.QueryInterface(iodev, ref deviceInterface, out devPtr);
```

Comment:
Was hoping that the last reference was right, but apparently MacOS doesn't support COM Interop
```
Exception has occurred: CLR/System.PlatformNotSupportedException
Exception thrown: 'System.PlatformNotSupportedException' in System.Private.CoreLib.dll: 'COM Interop is not supported on this platform.'
   at System.Runtime.InteropServices.Marshal.QueryInterface(IntPtr pUnk, Guid& iid, IntPtr& ppv)
   at HidSharp.Platform.MacOS.MacHidDevice.GetDeviceString(Int32 index) in /Users/infinity/Documents/Repos/HidSharpCore/HidSharp/Platform/MacOS/MacHidDevice.cs:line 209
```